src/vmm/src/vstate/kvm.rs

```diff
@@ -86,6 +98,74 @@ impl Kvm {
         }
     }
 
+    #[cfg(target_arch = "x86_64")]
+    // XSTATE feature mask for Intel AMX.
+    const INTEL_AMX_XCOMP_MASK: libc::c_ulong =
+        (1u64 << arch_prctl::ARCH_XCOMP_TILECFG) | (1u64 << arch_prctl::ARCH_XCOMP_TILEDATA);
+
+    /// Enable Intel AMX if available.
+    ///
+    /// Intel AMX (Advanced Matrix Extensions) is an instruction set for AI workloads that was
+    /// introduced in Intel Sapphire Rapids (*7i.metal). Since it requires larger area to save the
+    /// state, it is disabled by default.
+    /// https://github.com/torvalds/linux/blob/master/Documentation/arch/x86/xstate.rst
+    ///
+    /// We enable it by default but can be disabled by CPU template; otherwise,
+    /// KVM_GET_SUPPORTED_CPUID returns a inconsistent state where TILECFG is enabled but TILEDATA
+    /// is disabled, causing guest's #GP fault on xsetbv due to the lack of sanity check.
+    /// https://lore.kernel.org/all/20230405004520.421768-1-seanjc@google.com/
```

Comment:
Will this fix get backported to 6.1? 

Also do I understood it right that this is then only safe to disable via CPU template if we are using a v6.4+ kernel?

Reply:
Regardless of whether backporting or not, we have to request permission of XTILEDATA.

CPU template can disable it safely on any kernels supporting Intel AMX.

The issue that was fixed in v6.4 is that we cannot directly pass the value returned from KVM_GET_SUPPORTED_CPUID to KVM_SET_CPUID2 without the arch_prctl() call. As long as we call arch_prctl() before KVM_GET_SUPPORTED_CPUID, the issue doesn't affect us.